src/data-api/data-api-client.js

```diff
@@ -0,0 +1,755 @@
+      ? args[0].sql
+      : error(`No 'sql' statement provided.`);
+
+// Parse the parameters from provided arguments
+const parseParams = args =>
+  Array.isArray(args[0].parameters)
+    ? args[0].parameters
+    : typeof args[0].parameters === 'object'
+      ? [args[0].parameters]
+      : Array.isArray(args[1])
+        ? args[1]
+        : typeof args[1] === 'object'
+          ? [args[1]]
+          : args[0].parameters
+            ? error("'parameters' must be an object or array")
+            : args[1]
+              ? error('Parameters must be an object or array')
+              : [];
+
+// Parse the supplied database, or default to config
+const parseDatabase = (config, args) =>
+  config.transactionId
+    ? config.database
+    : typeof args[0].database === 'string'
+      ? args[0].database
+      : args[0].database
+        ? error("'database' must be a string.")
+        : config.database
+          ? config.database
+          : undefined; // removed for #47 - error('No \'database\' provided.')
+
+// Parse the supplied hydrateColumnNames command, or default to config
+const parseHydrate = (config, args) =>
+  typeof args[0].hydrateColumnNames === 'boolean'
+    ? args[0].hydrateColumnNames
+    : args[0].hydrateColumnNames
+      ? error("'hydrateColumnNames' must be a boolean.")
+      : config.hydrateColumnNames;
+
+// Parse the supplied format options, or default to config
+const parseFormatOptions = (config, args) =>
+  typeof args[0].formatOptions === 'object'
+    ? {
+      deserializeDate:
+        typeof args[0].formatOptions.deserializeDate === 'boolean'
+          ? args[0].formatOptions.deserializeDate
+          : args[0].formatOptions.deserializeDate
+            ? error("'formatOptions.deserializeDate' must be a boolean.")
+            : config.formatOptions.deserializeDate,
+      treatAsLocalDate:
+        typeof args[0].formatOptions.treatAsLocalDate === 'boolean'
+          ? args[0].formatOptions.treatAsLocalDate
+          : args[0].formatOptions.treatAsLocalDate
+            ? error("'formatOptions.treatAsLocalDate' must be a boolean.")
+            : config.formatOptions.treatAsLocalDate
+    }
+    : args[0].formatOptions
+      ? error("'formatOptions' must be an object.")
+      : config.formatOptions;
+
+
+// Utility function for removing certain keys from an object
+const omit = (obj, values) =>
+  Object.keys(obj).reduce(
+    (acc, x) =>
+      values.includes(x) ? acc : Object.assign(acc, { [x]: obj[x] }),
+    {}
+  );
+
+// Prepare method params w/ supplied inputs if an object is passed
+const prepareParams = ({ secretArn, resourceArn }, args) => ({
+    secretArn,
+      resourceArn,
+    ...(typeof args[0] === 'object'
+      ? omit(args[0], ['hydrateColumnNames', 'parameters'])
+      : {}) // merge any inputs
+  });
+
+// Utility function for picking certain keys from an object
+const pick = (obj, values) =>
+  Object.keys(obj).reduce(
```

Comment:
Weird indent here in the function...

Reply:
I let eslint take care of formatting but it ruins the formatting. So manually formatted the ternaries.